Plugins/IngameDebugConsole/Scripts/DebugLogConsole.cs

```diff
@@ -376,6 +369,29 @@ public static void AddCustomParameterType( Type type, ParseFunction parseFunctio
 				typeReadableNames[type] = typeReadableName;
 		}
 
+		public static void AddCustomParameterType(MethodInfo method, Type type, string readableName)
+		{
+			if (TryBuildCustomParameterFunction(method, out ParseFunction func, false))
+				AddCustomParameterType(type, func, readableName);
+		}
+
+		private static bool TryBuildCustomParameterFunction(MethodInfo method, out ParseFunction function, bool validate)
+		{
+			try
+			{
+				function = (ParseFunction)Delegate.CreateDelegate(typeof(ParseFunction), method);
+				return true;
+			}
+			catch (Exception e)
+			{
+				const string format = "Parser Method {0}.{1} is Invalid.\n{2}\nex: public static bool {1}(string input, out object result)";
+				string error = string.Format(format, method.DeclaringType.FullName, method.Name, e.Message);
+				Debug.LogError(error);
+				function = null;
+				return false;
+			}
+		}
+
```

Comment:
TryBuildCustomParameterFunction isn't called elsewhere so could you combine these two functions?

Reply:
Do we want to break when an exception is throw or just to skip this one method? If we don't catch here, all the next methods will be ignored.

Reply:
I'm good with an unhandled exception in this case. There is an error that must be resolved and in both approaches, that error will be logged as soon as the developer enters Play mode. An exception will force the developer to resolve the issue which is a win-win in my book: it's simpler and more effective.